src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -272,9 +272,9 @@ public boolean canProvision(Label label) {
     // item in the list of configured Mesos labels.
     // TODO(vinod): The framework may not have the resources necessary
     // to start a task when it comes time to launch the slave.
-    if (label != null && slaveInfos != null) {
+    if (slaveInfos != null) {
       for (MesosSlaveInfo slaveInfo : slaveInfos) {
-        if (label.matches(Label.parse(slaveInfo.getLabelString()))) {
+        if (slaveInfo.matchesLabel(label)) {
```

Comment:
Just to confirm, does this still work in a non-mesos only jenkins installation? i.e., if all of the mesos slaves are configured with a non-null label, do jobs without labels run on jenkins master or non-mesos jenkins slaves?